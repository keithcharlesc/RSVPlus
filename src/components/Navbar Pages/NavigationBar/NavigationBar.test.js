import React from "react";
import MockNavigationBar from "./MockNavigationBar";
import { render, getByTestId } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { toHaveTextContent } from "@testing-library/jest-dom";

it("renders without crashing, has correct text, and correct urls for navigation tabs", () => {
  const { getByText } = render(<MockNavigationBar />);
  getByText("Home");
  expect(getByText("Home").href).toBe("http://localhost/");
  getByText("Channels");
  expect(getByText("Channels").href).toBe("http://localhost/channels");
  getByText("Calendar");
  expect(getByText("Calendar").href).toBe("http://localhost/calendar");
  getByText("FAQ");
  expect(getByText("FAQ").href).toBe("http://localhost/faq");
  getByText("Contact Us");
  expect(getByText("Contact Us").href).toBe("http://localhost/contact-us");
  getByText("Signed in as: Bob");
  expect(getByTestId(document.documentElement, "button")).toHaveTextContent(
    "Log Out"
  );
});

it("matches snapshot", () => {
  const tree = TestRenderer.create(<MockNavigationBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
