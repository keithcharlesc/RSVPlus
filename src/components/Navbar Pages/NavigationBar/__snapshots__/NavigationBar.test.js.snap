// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="nav-bar-appName text-danger navbar-brand"
      href="/"
    >
      RSVP+
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-bar-text px-3 nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-bar-text px-3 nav-link"
          data-rb-event-key="/channels"
          href="/channels"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Channels
        </a>
        <a
          className="nav-bar-text px-3 nav-link"
          data-rb-event-key="/calendar"
          href="/calendar"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Calendar
        </a>
        <a
          className="nav-bar-text px-3 nav-link"
          data-rb-event-key="/faq"
          href="/faq"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          FAQ
        </a>
        <a
          className="nav-bar-text px-3 nav-link"
          data-rb-event-key="/contact-us"
          href="/contact-us"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Contact Us
        </a>
      </div>
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <span
          className="nav-bar-name navbar-text"
        >
          Signed in as: 
          Bob
          <img
            className="photo ml-2 rounded-circle"
            src="thttp://email.com/image"
          />
        </span>
      </div>
    </div>
    <button
      className="ml-3 btn btn-outline-primary"
      data-testid="button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Log Out
    </button>
  </nav>
  <div>
    
  </div>
</div>
`;
